### ISSUE TITLE
Excessive output from Mac Aarch64 builds (TRSS can not parse results for Java 17 release)

### ISSUE THREAD

**Issue by karianna** [adoptium/temurin-build#2835]
https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk17-pipeline/188/ is the pipeline that kicked off the build

https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk17/job/jdk17-mac-aarch64-hotspot/46/console is the build job that then triggers the test job (which all completed)

NOTE: The test runs were delayed due to an offline machine, but once switched on it went to completion.

**Comment by smlambert** [comment]
This issue has been transferred from aqa-test-tools repo because the underlying cause of not being able to display mac aarch64 in TRSS is due to the enormous amount of output from the build job.  Need to look at why the build for this platform is so noisy and fix it.

**Comment by adamfarley** [comment]
Things tried (TLDR at the bottom):
- Checked different versions on the same platform.
  - JDK11u output is under 1mb.
  - JDK17u output is about 170mb
  - JDK18u output is about 90mb (still above TRSS' limit)
- Checked the config command for obvious differences between jdk11, 17, and 18. There are three: 
  1) The boot JDK differs.
    - JDK11u: /Users/jenkins/jdk-11.0.15+8/Contents/Home
    - JDK17u: /Users/jenkins/zulu16.28.11-ca-jdk16.0.0-macosx_aarch64/zulu-16.jdk/Contents/Home
    - JDK18u: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home 
  2) JDK11u has "--with-jvm-features=shenandoahgc". The others do not.
  3) JDK11u has "--disable-ccache". The others do not.
    - Tried re-running JDK17u with this option. The output remains the same size.
- Checked the JDK17u output for things producing the most lines.
  - There are 1319151 lines of output, with a total size of 170mb.
    - This makes about 7,759 lines per mb.
    - Given the TRSS limit (~50mb), this means we need to lose a minimum of 931k lines.
      - About 189636 instances of unused parameter warnings. 
        - 3 lines each makes 568,908 lines total.
        - These can be suppressed via the native compiler option -Wno-unused-parameter
      - About 26281 instances of unused local typedefs.
        - 4 lines each makes 105,124 lines total.
        - These can be suppressed via the native compiler option -Wno-unused-local-typedefs
      - Testing our ability to suppress the two warnings above via "with-extra-cflags" config arg.
      - About 112073 instances of "note: expanded from macro", with various macros.
        - 4 lines each makes 448,292 lines total.
        - I'm unable to find a way to easily suppress these. 
          - Nor have I found a clear explanation of why this is a warning. 
          - Note: Some sites suggest that this is just supplimentary information to the error itself. Since the instances I see come after the unused parameter warnings, it's possible that suppressing that issue will suppress these as well.
      - Here is the proposed "fix" for suppressing the above errors:
        - "CONFIGURE_ARGS": "--with-extra-cxxflags='-Wno-unused-local-typedefs -Wno-unused-parameter'"
        - Trial run: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-aarch64-temurin/33/


TLDR: Fix trialled [here](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-aarch64-temurin/33/). If the console output is under 50mb in size, then the "with-extra-cxxflags" config addition should be added to the default arguments for this platform.

**Comment by smlambert** [comment]
Remains unresolved for [April release](https://trss.adoptium.net/output/build?id=625f2bc472761a4c5c13bbf3).  We will want to come back to this post-release to resolve.
<img width="758" alt="Screen Shot 2022-04-26 at 6 28 10 AM" src="https://user-images.githubusercontent.com/2836948/165280725-049e7ff1-5d20-4ead-be5e-f7c1b2a0af25.png">

**Comment by adamfarley** [comment]
After reviewing with Andrew, the fundamental issue was determined to be in flags-cflags.m4. For clang (the c compiler we use on mac aarch) the value of BUILD_CC_DISABLE_WARNING_PREFIX is not set to "-Wno-", thus leaving the value blank when buildjdk-spec.gmk is created from buildjdk-spec.gmk.in.

Test build running [here](https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-mac-aarch64-temurin/53/), though Andrew has already verified locally.

Once we have Jenkins confirmation as well, a patch will be created for the upstream head repository and backported to jdk17u.

**Comment by adamfarley** [comment]
It works. Will create an upstream bug and a PR after lunch, followed by a backport request for 17 and 18.

**Comment by smlambert** [comment]
Nice!  This made a big difference on being able to see this output in TRSS now, thanks!

**Comment by sxa** [comment]
Thanks for confirming Shelley! I hadn't made it back round the loop to check TRSS yet.
